# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/power/WakeLockJS.cpp
# Commit: f3c72b5730df
# Full Hash: f3c72b5730df88fce5472439458ad122f9a2d83e
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2023-12-06 05:18:37
# Regressor Bug: 1589554
# File Overlap Count: 1
# Description:
#   Bug 1589554 - Part 7: Add logging to Screen Wake Lock. r=dom-core,edgar
#   
#   Depends on D189513
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D189514
# ==============================================================================

diff -r 4c6403ab4174 -r f3c72b5730df dom/power/WakeLockJS.cpp
--- a/dom/power/WakeLockJS.cpp	Tue Dec 05 23:58:08 2023 +0000
+++ b/dom/power/WakeLockJS.cpp	Tue Dec 05 23:58:08 2023 +0000
@@ -7,6 +7,7 @@
 #include "ErrorList.h"
 #include "mozilla/AlreadyAddRefed.h"
 #include "mozilla/Assertions.h"
+#include "mozilla/Logging.h"
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/Event.h"
 #include "mozilla/dom/EventTarget.h"
@@ -31,6 +32,8 @@
 
 namespace mozilla::dom {
 
+static mozilla::LazyLogModule sLogger("ScreenWakeLock");
+
 #define MIN_BATTERY_LEVEL 0.05
 
 nsLiteralCString WakeLockJS::GetRequestErrorMessage(RequestError aRv) {
@@ -104,6 +107,7 @@
   RefPtr<WakeLockSentinel> kungFuDeathGrip = aLock;
   aDoc->ActiveWakeLocks(aType).Remove(aLock);
   aLock->NotifyLockReleased();
+  MOZ_LOG(sLogger, LogLevel::Debug, ("Released wake lock sentinel"));
 }
 
 NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE(WakeLockJS, mWindow)
@@ -160,6 +164,7 @@
 // https://w3c.github.io/screen-wake-lock/#the-request-method
 already_AddRefed<Promise> WakeLockJS::Request(WakeLockType aType,
                                               ErrorResult& aRv) {
+  MOZ_LOG(sLogger, LogLevel::Debug, ("Received request for wake lock"));
   nsCOMPtr<nsIGlobalObject> global = mWindow->AsGlobal();
 
   RefPtr<Promise> promise = Promise::Create(global, aRv);
@@ -181,6 +186,8 @@
         if (lockOrErr.isOk()) {
           RefPtr<WakeLockSentinel> lock = lockOrErr.unwrap();
           promise->MaybeResolve(lock);
+          MOZ_LOG(sLogger, LogLevel::Debug,
+                  ("Resolved promise with wake lock sentinel"));
         } else {
           promise->MaybeRejectWithNotAllowedError(
               GetRequestErrorMessage(lockOrErr.unwrapErr()));
